src/overlays/actors/ovl_En_Bb/z_en_bb.c

```diff
@@ -36,118 +47,667 @@ const ActorInit En_Bb_InitVars = {
+            this->collider.base.atFlags &= ~AT_HIT;
+            this->actor.world.rot.y = this->actor.yawTowardsPlayer + 0x8000;
+            this->actor.shape.rot.y = this->actor.world.rot.y;
+            Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_BUBLE_BITE);
+
+            if (this->flameScaleX > 0.0f) {
+                gSaveContext.jinxTimer = 1200;
+            }
+
+            if (this->actionFunc == EnBb_Attack) {
+                EnBb_SetupFlyIdle(this);
+            }
+        }
+    }
+}
+
+void EnBb_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnBb* this = THIS;
+
+    EnBb_UpdateDamage(this, globalCtx);
+    this->actionFunc(this, globalCtx);
+    if ((this->actionFunc != EnBb_Dead) && (this->actionFunc != EnBb_WaitForRevive)) {
+        Actor_MoveWithGravity(&this->actor);
+        Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 30.0f, 25.0f, 40.0f, 7U);
+
+        this->collider.dim.worldSphere.center.x = this->actor.world.pos.x;
+        this->collider.dim.worldSphere.center.y = this->actor.world.pos.y + 15.0f;
+        this->collider.dim.worldSphere.center.z = this->actor.world.pos.z;
+        this->collider.dim.worldSphere.radius = this->flameScaleX * 30.0f;
+        this->collider.dim.worldSphere.radius = CLAMP_MIN(this->collider.dim.worldSphere.radius, 20);
+
+        Math_Vec3s_ToVec3f(&this->actor.focus.pos, &this->collider.dim.worldSphere.center);
+
+        if (this->collider.base.atFlags & AT_ON) {
+            this->actor.flags |= ACTOR_FLAG_1000000;
+            CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        }
+
+        if (this->collider.base.acFlags & AC_ON) {
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        }
+
+        if (this->collider.base.ocFlags1 & OC1_ON) {
+            CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        }
+
+        if (this->drawDmgEffAlpha > 0.0f) {
+            if (this->drawDmgEffType != ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX) {
+                Math_StepToF(&this->drawDmgEffAlpha, 0.0f, 0.05f);
+                this->drawDmgEffScale = (this->drawDmgEffAlpha + 1.0f) * 0.2f;
+                this->drawDmgEffScale = CLAMP_MAX(this->drawDmgEffScale, 0.4f);
+            } else if (!Math_StepToF(&this->drawDmgEffFrozenSteamScale, 0.4f, 0.01f)) {
+                func_800B9010(&this->actor, NA_SE_EV_ICE_FREEZE - SFX_FLAG);
+            }
+        }
+    }
+}
+
+s32 EnBb_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnBb* this = THIS;
+
+    if (this->bodyPartDrawStatus == BB_BODY_PART_DRAW_STATUS_BROKEN) {
+        this->limbDList = *dList;
+        *dList = NULL;
+    }
+
+    return false;
+}
+
+void EnBb_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    s32 pad;
+    EnBb* this = THIS;
+    MtxF* temp_v0_4;
```

Comment:
```suggestion
    MtxF* mtx;
```
Should probably name this temp. I'd go with `mtx`, but up to you. Down below looks like you did something like `currentMatrixState` for something similar.